tests/html/cloned-template-script.html

```diff
@@ -0,0 +1,85 @@
+            template = importLink.import.querySelector('template');
+            done();
+          });
+        });
+
+        teardown(function() {
+          window.executedTemplateScript = undefined;
+          window.executedNestedTemplateScript = undefined;
+          container.innerHTML = '';
+        });
+
+        test('import does not execute scripts in templates', function() {
+          assert.isUndefined(window.executedTemplateScript, 'remote template script is NOT executed after import');
+          assert.isUndefined(window.executedNestedTemplateScript, 'remote nested template script is NOT executed after import');
+        });
+
+        test('executes scripts when template is cloned and attached', function() {
+          container.appendChild(document.importNode(template.content, true));
+          assert.equal(window.executedTemplateScript, 1, 'remote template script executed once after clone and attach to the main document');
+          assert.isUndefined(window.executedNestedTemplateScript, 'remote nested template script is NOT executed after import');
+        });
+
+        test('executes scripts when nested template of cloned template is cloned and attached', function() {
+          container.appendChild(document.importNode(template.content, true));
+          var nestedTemplate = container.querySelector('template');
+          container.appendChild(document.importNode(nestedTemplate.content, true));
+          assert.equal(window.executedNestedTemplateScript, 1, 'remote nested template script executed after clone and attach to the main document');
+        });
+
+        test('executes scripts of each clone', function() {
+          container.appendChild(document.importNode(template.content, true));
+          assert.equal(window.executedTemplateScript, 1, 'remote template script executed after clone and attach to the main document');
+          container.appendChild(document.importNode(template.content, true));
+          assert.equal(window.executedTemplateScript, 2, 'remote template script executed after 2nd clone and attach to the main document');
+          assert.isUndefined(window.executedNestedTemplateScript, 'remote nested template script is NOT executed after parent is clone and attach to the main document');
+        });
+
+        // NOTE: these tests MUST be kept last, since once the `template.content` is attached,
+        // it will be an empty DocumentFragment in the next tests.
+        test('executes scripts when nested template is cloned and attached', function() {
+          var nestedTemplate = template.content.querySelector('template');
+          container.appendChild(document.importNode(nestedTemplate.content, true));
+          assert.equal(window.executedNestedTemplateScript, 1, 'remote nested template script executed after clone and attach to the main document');
+        });
+
+        test('executes scripts when template is attached', function() {
```

Comment:
This test should be removed, as no browser executes scripts generated via `template.innerHTML`, the user needs to clone the template in order to execute its scripts